### PR TITLE
✨ Support <amp-ima-video dock>

### PR DESCRIPTION
**Pull request by alanorozco**
## Breakdown

- ✅ Validate `<amp-ima-video dock>`
- ✨ Support `data-poster` for docking placeholder per `amp-ima-video`'s API

## 🐛 Known issues

- Two sets of controls while docked video is paused. 
  /cc @torch2424 Is it possible to add an API to disable controls in `amp-ima-video` entirely?

Partial for #20759

### CODE REVIEW

File: extensions/amp-ima-video/0.1/test/validator-amp-ima-video-dock-no-extension.html [hunk 1]
```diff
@@ -0,0 +1,40 @@
+<!--
```
Comment (aghassemi) [inline extensions/amp-ima-video/0.1/test/validator-amp-ima-video-dock-no-extension.html]: Let's update https://github.com/ampproject/amphtml/blob/master/spec/amp-video-interface.md and remove experimental wording from it as well.

Also update all the supported video's `.md` files with the `dock` attribute similar to autoplay.

Comment (aghassemi) [reply extensions/amp-ima-video/0.1/test/validator-amp-ima-video-dock-no-extension.html]: (separate PR okay)

**Comment by torch2424** [comment]
@alanorozco Yes this is definitely possible. Feel free to open an issue, and assign it to me 😄

File: extensions/amp-ima-video/0.1/test/validator-amp-ima-video-dock-no-extension.html (hunk 1 above)
Comment (alanorozco) [reply extensions/amp-ima-video/0.1/test/validator-amp-ima-video-dock-no-extension.html]: Cool, #20788

**Pull request merged**